app/src/main/java/com/github/sdpteam15/polyevents/view/activity/TimeTableActivity.kt

```diff
@@ -0,0 +1,515 @@
+        constraintLayout.addView(eventView2)
+        displayedViews.add(eventView2)
+        idViewToIdEvent[currentId] = event.eventId!!
+
+        eventView2.setOnClickListener {
+            val intent = Intent(applicationContext, EventActivity::class.java).apply {
+                putExtra(EXTRA_EVENT_ID, event.eventId)
+            }
+            startActivity(intent)
+        }
+
+        //Clone the current constraints of the constraintLayout
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+
+        //Compute the margin top to have the view start at the right position to indicate start time
+        val marginTop = ((start.minute / 60.0) * hourSizeDp).toInt()
+
+        //Constraints for the time
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.LEFT,
+            idLine,
+            ConstraintSet.LEFT,
+            0
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.RIGHT,
+            idLine,
+            ConstraintSet.RIGHT,
+            0
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.TOP,
+            idLine,
+            ConstraintSet.BOTTOM,
+            marginTop.dpToPixelsInt(this)
+        )
+        constraintSet.applyTo(constraintLayout)
+    }
+
+    /**
+     * Adds the line that indicates the current time
+     */
+    private fun addNowBar() {
+        val now = LocalDateTime.now()
+        val nowH = now.hour
+        val nowM = now.minute
+        val idLine = hourToLine[nowH] ?: return
+
+        val currentId = nextId++
+        val constraintLayout = findViewById<ConstraintLayout>(R.id.id_timetable_constraintlayout)
+
+        //Create now line
+        val line = View(this)
+        line.backgroundTintList = resources.getColorStateList(R.color.red, null)
+        line.id = currentId
+
+        //Create new parameters for the line (the line)
+        val params = ViewGroup.LayoutParams(0, nowLineHeightDP.dpToPixelsInt(this))
+        line.layoutParams = params
+        line.setBackgroundColor(Color.BLUE)
+
+        constraintLayout.addView(line)
+        nowBar = line
+
+
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+        val marginTop = ((nowM / 60.0) * hourSizeDp).toInt()
+        //Constraints for the line
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.TOP,
+            idLine,
+            ConstraintSet.BOTTOM,
+            marginTop.dpToPixelsInt(this)
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.LEFT,
+            idLine,
+            ConstraintSet.LEFT,
+            linepaddingLeftDP.dpToPixelsInt(this)
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.RIGHT,
+            constraintLayout.id,
+            ConstraintSet.RIGHT,
+            0
+        )
```

Comment:
Another suggestion but maybe it will be easier to create a function that takes currentid as input and one or two parameters since you often call three connect in a row with the same "constraint places"

Reply:
I can't abstract this as I would need to give all elements as parameters because they slightly change every time I call them